sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -664,19 +669,61 @@ func tagImage(ctx context.Context, imageName string, newImageName string, client
 
 // withDynamicResolver provides an initialized resolver for use with ref.
 func withDynamicResolver(ctx context.Context, ref string) containerd.RemoteOpt {
-	if !strings.HasPrefix(ref, "ecr.aws/") {
-		// not handled here
-		return func(_ *containerd.Client, _ *containerd.RemoteContext) error { return nil }
-	}
+	if strings.HasPrefix(ref, "ecr.aws/") {
+		// For ECR registries, we need to use the Amazon ECR resolver
+		return func(_ *containerd.Client, c *containerd.RemoteContext) error {
+			// Create the Amazon ECR resolver
+			resolver, err := ecr.NewResolver()
+			if err != nil {
+				return errors.Wrap(err, "Failed to create ECR resolver")
+			}
+			log.G(ctx).WithField("ref", ref).Info("pulling with Amazon ECR Resolver")
+			c.Resolver = resolver
+			return nil
+		}
+	} else if strings.HasPrefix(ref, "public.ecr.aws") {
+		// For Amazon ECR Public registries, we need to fetch credentials before resolving the image reference
+		// TODO Potentially need to implement credentials caching to avoid throttling during large deployments
```

Comment:
If we do credential caching, we'll need a bit more of a design for handling those credentials securely.  It might be worth removing this comment and opening an issue for it instead.